artnet/__init__.py

```diff
@@ -25,32 +26,137 @@
+
+    def get_port(self):
+        """
+        Returns the port of the ArtNet node
+
+        :return: port of the device, as set in plugin.conf
+        """
+        return self._port
+
+    def get_universe(self):
+        """
+        Returns the Universe of the ArtNet node
+
+        :return: Universe of the device, as set in plugin.conf
+        """
+        return self._universe
+
+    def get_net(self):
+        """
+        Returns the net of the ArtNet node
+
+        :return: net of the device, as set in plugin.conf
+        """
+        return self._net
+        
+    def get_subnet(self):
+        """
+        Returns the Subnet of the ArtNet node
+
+        :return: Subnet of the device, as set in plugin.conf
+        """
+        return self._subnet
+        
+    def get_items(self):
+        """
+        Returns added items
+
+        :return: array of items held by the device
+        """
+        return self._items
 
 class ArtNet(SmartPlugin):
 
-    ALLOW_MULTIINSTANCE = False
-    PLUGIN_VERSION = '1.3.0'
+    ALLOW_MULTIINSTANCE = True
+    PLUGIN_VERSION = "1.5.1"
+    ADDR_ATTR = 'artnet_address'
 
     packet_counter = 1
     dmxdata = [0, 0]
+    
+    def __init__(self, sh, *args, **kwargs):
+        """
+        Initalizes the plugin. The parameters describe for this method are pulled from the entry in plugin.conf.
+        """
+        # self.logger = logging.getLogger(__name__)
+        self.logger.info('Init ArtNet Plugin')
+        
+        self._model = ArtNet_Model(self.get_parameter_value('ip') or '127.0.0.1', 
+                                   port=int(self.get_parameter_value('port') or 6454),
+                                   universe=int(self.get_parameter_value('artnet_universe') or 0),
+                                   net=int(self.get_parameter_value('artnet_net') or 0),
+                                   subnet=int(self.get_parameter_value('artnet_subnet') or 0),
+                                   instance_name=self.get_instance_name(),
+                                   update_cycle=int(self.get_parameter_value('update_cycle') or 0),
+                                   )
 
-    def __init__(self, smarthome, artnet_net, artnet_subnet, artnet_universe, ip, port):
-        self.logger = logging.getLogger(__name__)
         self.s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        self.net = int(artnet_net)
-        self.subnet = int(artnet_subnet)
-        self.universe = int(artnet_universe)
-        self.ip = ip
-        self.port = int(port)
-        self.logger.debug("Init ArtNet Plugin done")
+        self.init_webinterface()
+
+        self.logger.debug("Init ArtNet Plugin for %s done" % self._model._instance_name)
 
+    def parse_item(self, item):
+        # items bound to this artnet-subnet
+        if self.has_iattr(item.conf, self.ADDR_ATTR):
+            adr = int(self.get_iattr_value(item.conf, self.ADDR_ATTR))
+            if adr > 0 and adr < 513:
```

Comment:
I think this condition can be move into the `plugin.yaml` by using `valid_min` & `valid_max` (see [docs](https://www.smarthomeng.de/developer/metadata/item_attributes.html)).

Reply:
Is this check based on yaml enforced? If so: fine! If not: it should stay here (in addition to the definition in the yaml).

Reply:
Was added to the yaml...

Reply:
i think u can then remove the check from your code. it should be directly checked by the yaml information

Reply:
Is this check as well done once the yaml is edited with a normal text-editor? My code is throwing an error if I try to use address 123456 - not a general validation engine.

Reply:
i think shng needs to be restarted for that. did that fix it?